public/app/plugins/datasource/postgres/datasource.ts

```diff
@@ -123,27 +123,27 @@ export class PostgresDatasource {
       .then(data => this.responseParser.parseMetricFindQueryResult(refId, data));
   }
 
-  testDatasource() {
-    return this.backendSrv
-      .datasourceRequest({
-        url: '/api/tsdb/query',
-        method: 'POST',
-        data: {
-          from: '5m',
-          to: 'now',
-          queries: [
-            {
-              refId: 'A',
-              intervalMs: 1,
-              maxDataPoints: 1,
-              datasourceId: this.id,
-              rawSql: 'SELECT 1',
-              format: 'table',
-            },
-          ],
-        },
-      })
+  testDatasource(control) {
+    return this.metricFindQuery('SELECT 1', {})
       .then(res => {
+        if (control.current.jsonData.timescaledb === 'auto') {
```

Comment:
See earlier comment regarding graphite datasource. Feels a bit weird to make changes in the test datasource function.